### PR TITLE
[INLONG-9858][Agent] Increase local read and write capabilities for module config

### LINKED ISSUE #9858: [Improve][Agent] Increase local read and write capabilities for module config
### Description

Increase local read and write capabilities for module config

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justinwwhuang**
[INLONG-9858][Agent] Increase local read and write capabilities for module config
- Fixes #9858

### Motivation
Increase local read and write capabilities for module config

### Modifications

Increase local read and write capabilities for module config

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java [hunk 1]
```diff
@@ -106,51 +120,110 @@ public String getCurrentMd5() {
...
+                        currentModules.put(module.getId(), module);
+                    });
+                } else {
+                    LOGGER.info("modules in local file invalid");
+                }
+            } else {
+                LOGGER.info("local module json file new FileInputStream failed");
+            }
+        } catch (FileNotFoundException e) {
+            LOGGER.info("local module json file not found");
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:172]: LOGGER.warn or LOGGER.error

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java [hunk 2]
```diff
@@ -106,51 +120,110 @@ public String getCurrentMd5() {
...
         };
     }
 
+    public void restoreFromLocalFile(String confPath) {
+        LOGGER.info("restore modules from local file");
+        Reader reader = null;
+        try {
+            InputStream inputStream = new FileInputStream(confPath + LOCAL_CONFIG_FILE);
+            if (inputStream != null) {
+                reader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:157]: It's recommand to use  try-with-resource for `Reader` as the `saveToLocalFile()` method does

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java [hunk 3]
```diff
@@ -106,51 +120,110 @@ public String getCurrentMd5() {
...
-     * thread for core thread.
+     * Thread for core thread.
      *
      * @return runnable profile.
      */
     private Runnable coreThread() {
         return () -> {
-            Thread.currentThread().setName("task-manager-core");
+            Thread.currentThread().setName("module-manager-core");
+            restoreFromLocalFile(confPath);
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:135]: One thing I am concerned about is whether the result of `restoreFromLocalFile` has effects on `dealWithConfigQueue`.
i.e. does the `dealWithConfigQueue` method be invoked even if the `restoreFromLocalFile` finished in exception?

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java [hunk 4]
```diff
@@ -81,8 +95,8 @@ public HttpManager getHttpManager(InstallerConfiguration conf) {
                 DEFAULT_AGENT_MANAGER_VIP_HTTP_PREFIX_PATH);
         int timeout = conf.getInt(AGENT_MANAGER_REQUEST_TIMEOUT,
                 DEFAULT_AGENT_MANAGER_REQUEST_TIMEOUT);
-        String secretId = conf.get(MANAGER_AUTH_SECRET_ID);
-        String secretKey = conf.get(MANAGER_AUTH_SECRET_KEY);
+        String secretId = conf.get(MANAGER_AUTH_SECRET_ID, "");
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:98]: This modification changes the behavior.
Without this pull request, this line will throw an exception if the `secretId` is not configured, but nothing happens with this pull request.

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java (hunk 3 above)
Comment (justinwwhuang) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:135]: yes, dealWithConfigQueue method WILL be invoked even if the restoreFromLocalFile finished in exception

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java (hunk 1 above)
Comment (justinwwhuang) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:172]: Even if there are no files, it is considered normal

Comment (aloyszhang) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:172]: Got it! Thank you for your explanation.

**Review by aloyszhang — APPROVED** [review]
LGTM

**Review by haifxu — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**